### ISSUE TITLE
map_to_interpret_space function for Lime and Kernel Shap

### ISSUE THREAD

**Issue by dejeanph** [deel-ai/xplique#92]
The methods Lime and KernelShap need a map_to_interpret_space function. By default, the function is the quickshift segmentation algorithm. For better results, watershed function on images have to be used. This algorithm, in the skimage, package needs a parameter (makers) which value is dedicated to the image treated  and not for all images of the dataset. 
The current API of Lime and KernelShap cannot allowed such configuration (function + dedicated value). 

A solution could be to have a parameter for the image maps computed in advance instead of compute them every time in the explicability method function.

**Comment by lucashervier** [comment]
You want to change the default `map_to_interpret_space` function for LIME & KernelSHAP for images inputs to watershed. But  from my understanding this new function need to be parametrized for EACH images which I do not believe make the utilisation of the library easier for someone beginning with XAI. Therefore, I do not believe that changing  the default function is a good solution.

However, what we can do is to propose a pool of image segmentation algorithms including watershed. Since I am not familiar with watershed can you provide some of your workaround (in a colab notebook) in order to know exactly what you expect and how you are currently doing ? 

Thanks!

**Comment by RuoyuChen10** [comment]
> You want to change the default `map_to_interpret_space` function for LIME & KernelSHAP for images inputs to watershed. But from my understanding this new function need to be parametrized for EACH images which I do not believe make the utilisation of the library easier for someone beginning with XAI. Therefore, I do not believe that changing the default function is a good solution.
> 
> However, what we can do is to propose a pool of image segmentation algorithms including watershed. Since I am not familiar with watershed can you provide some of your workaround (in a colab notebook) in order to know exactly what you expect and how you are currently doing ?
> 
> Thanks!

How to write a function? Can you give an example, please?

**Comment by AntoninPoche** [comment]
I agree that those methods' results are highly dependent on the segmentation. However, I did not see in the literature how to include the model in the segmentation, if you saw something and think it is necessary for the method, feel free to contribute.

However, we should keep in mind that `Lime` and `KernelShap` are black box methods, hence, the model's convolution layers cannot be taken into account, otherwise, it becomes a white box method.
